### PR TITLE
PERF: use is_foo_dtype fastpaths

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/arrays/masked.py [hunk 1]
```diff
@@ -156,7 +161,9 @@ def to_numpy(
         if na_value is lib.no_default:
             na_value = libmissing.NA
         if dtype is None:
-            dtype = object
+            dtype = np.dtype(object)
+        else:
+            dtype = pandas_dtype(dtype)
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/masked.py:166]: I don't think we should do this here, as this is converting to a numpy array, the `dtype` can never be a pandas dtype, and is always meant to be a numpy dtype (so let numpy interpret the string / dtype-like objects)

File: pandas/core/strings.py [hunk 2]
```diff
@@ -74,11 +74,11 @@ def cat_core(list_of_columns: List, sep: str):
     """
     if sep == "":
         # no need to interleave sep if it is empty
-        arr_of_cols = np.asarray(list_of_columns, dtype=object)
+        arr_of_cols = np.asarray(list_of_columns, dtype=np.dtype(object))
```
Comment (jorisvandenbossche) [inline pandas/core/strings.py:77]: You don't need to revert this to be clear, but I also don't think it's needed to change this in *numpy* functions

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good to me, just one small comment

File: pandas/core/strings.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/strings.py:77]: fair enough; i think in this case this was a find-and-replace

File: pandas/core/arrays/masked.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/masked.py:166]: ok

**Comment by jbrockmendel** [comment]
updated per comments

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**